### PR TITLE
Fix functional tests failed (android, push_file) 

### PR DESCRIPTION
**Pull request by ki4070ma**
For https://github.com/appium/python-client/issues/373

## Results

```
➜  android git:(fix-push-file-test) ✗ python2 remote_fs_tests.py         
test_pull_folder (__main__.RemoteFsTests) ... ok
test_push_file_with_src_path (__main__.RemoteFsTests) ... ok
test_push_pull_file (__main__.RemoteFsTests) ... ok

----------------------------------------------------------------------
Ran 3 tests in 73.231s

OK
➜  android git:(fix-push-file-test) ✗ python3 remote_fs_tests.py 
test_pull_folder (__main__.RemoteFsTests) ... ok
test_push_file_with_src_path (__main__.RemoteFsTests) ... ok
test_push_pull_file (__main__.RemoteFsTests) ... ok

----------------------------------------------------------------------
Ran 3 tests in 83.874s

OK
```

### CODE REVIEW

File: test/functional/android/remote_fs_tests.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+class RemoteFsTests(unittest.TestCase):
+    def setUp(self):
+        desired_caps = desired_capabilities.get_desired_capabilities('ApiDemos-debug.apk')
+        self.driver = webdriver.Remote('http://localhost:4723/wd/hub', desired_caps)
+
+    def tearDown(self):
+        self.driver.quit()
+
+        # remove zipped file from `test_pull_folder`
+        if hasattr(self, 'zipfilename') and os.path.isfile(self.zipfilename):
```
Comment (mykola-mokhnach) [inline test/functional/android/remote_fs_tests.py]: `if os.path.exists(getattr(self, 'zipfilename', ''))`

File: test/functional/android/remote_fs_tests.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        self.driver.push_file(path, base64.b64encode(data).decode('utf-8'))
+        data_ret = base64.b64decode(self.driver.pull_file(path))
+
+        self.assertEqual(data, data_ret)
+
+    def test_pull_folder(self):
+        string_data = b'random string data %d' % random.randint(0, 1000)
+        path = '/data/local/tmp'
+
+        self.driver.push_file(path + '/1.txt', base64.b64encode(string_data).decode('utf-8'))
```
Comment (mykola-mokhnach) [inline test/functional/android/remote_fs_tests.py]: use `os.path.join` to join paths

File: test/functional/android/remote_fs_tests.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        self.driver.push_file(path + '/2.txt', base64.b64encode(string_data).decode('utf-8'))
+
+        folder = self.driver.pull_folder(path)
+
+        # python doesn't have any functionality for unzipping streams
```
Comment (mykola-mokhnach) [inline test/functional/android/remote_fs_tests.py]: how about StringIO?

File: test/functional/android/remote_fs_tests.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        path = '/data/local/tmp/test_push_file.txt'
+        data = b'This is the contents of the file to push to the device.'
+
+        self.driver.push_file(path, base64.b64encode(data).decode('utf-8'))
+        data_ret = base64.b64decode(self.driver.pull_file(path))
+
+        self.assertEqual(data, data_ret)
+
+    def test_pull_folder(self):
+        string_data = b'random string data %d' % random.randint(0, 1000)
```
Comment (mykola-mokhnach) [inline test/functional/android/remote_fs_tests.py]: use `'{}'.format(var)` for formatting. After dropping python2 support I would use `f'{var}'` format

File: test/functional/android/remote_fs_tests.py (hunk 3 above)
Comment (ki4070ma) [reply test/functional/android/remote_fs_tests.py]: I didn't know StringIO, it's super cool.
But for here, I couldn't create zipfile object(here is `myzip`) with StringIO object.

File: test/functional/android/remote_fs_tests.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+            with open(src_path, 'rb') as fr:
+                original_data = fr.read()
+
+            self.driver.push_file(dest_path, source_path=src_path)
+            new_data = base64.b64decode(self.driver.pull_file(dest_path))
+            self.assertEqual(original_data, new_data)
+
+
+if __name__ == "__main__":
```
Comment (KazuCocoa) [inline test/functional/android/remote_fs_tests.py]: `'` (single quote) instead of `"` in this file?

Comment (ki4070ma) [reply test/functional/android/remote_fs_tests.py]: Yes.

I'll replace it in all cases  at this PR.

File: test/functional/android/remote_fs_tests.py (hunk 3 above)
Comment (mykola-mokhnach) [reply test/functional/android/remote_fs_tests.py]: hm, should be possible: https://stackoverflow.com/questions/10908877/extracting-a-zipfile-to-memory

Comment (mykola-mokhnach) [reply test/functional/android/remote_fs_tests.py]: https://stackoverflow.com/questions/5710867/downloading-and-unzipping-a-zip-file-without-writing-to-disk/42119017

Comment (ki4070ma) [reply test/functional/android/remote_fs_tests.py]: Thanks so much, I did.
(I didn't take care of decoding for zipfile data...)

File: test/functional/android/remote_fs_tests.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    def test_pull_folder(self):
+        data = appium_bytes('random string data {}'.format(random.randint(0, 1000)), 'utf-8')
+        dest_dir = '/data/local/tmp/'
+
+        for filename in ['1.txt', '2.txt']:
+            self.driver.push_file(os.path.join(dest_dir, filename), base64.b64encode(data).decode('utf-8'))
+
+        folder = self.driver.pull_folder(dest_dir)
+
+        filelist = ZipFile(BytesIO(base64.b64decode(folder))).namelist()
```
Comment (mykola-mokhnach) [inline test/functional/android/remote_fs_tests.py]: should we use ZipFile instance with `with` operator ?

Comment (ki4070ma) [reply test/functional/android/remote_fs_tests.py]: yes!

**Pull request merged**